qcodes/tests/test_measurement.py

```diff
@@ -501,146 +693,690 @@ def test_double_thread_measurement(self):
+
+    def test_step_out_multiparameter(self):
+        multi_parameter = MultiParameterTest("multi_param")
+
+        with Measurement('test_step_out_multiparameter') as msmt:
+            for k in Sweep(range(5), 'outer_sweep'):
+                for kk in Sweep(range(5), 'inner_sweep'):
+                    msmt.measure(multi_parameter)
+
+                    if kk > 2:
+                        msmt.step_out()
+                        break
+
+        verification_arrays = {
+            (0, 0, 0, 0): 1 * np.ones((5,5)),
+            (0, 0, 0, 1): 2 * np.ones((5,5)),
+            (0, 0, 0, 2): 3 * np.ones((5,5)),
+        }
+        verification_arrays[(0,0,0,0)][:,4:] = np.nan
+        verification_arrays[(0,0,0,2)][:,4:] = np.nan
+        verification_arrays[(0,0,0,1)][:,4:] = np.nan
+
+        verify_msmt(msmt, verification_arrays=verification_arrays, allow_nan=True)
+
+
+class ListHandler(logging.Handler):  # Inherit from logging.Handler
+    def __init__(self, log_list):
+        # run the regular Handler __init__
+        logging.Handler.__init__(self)
+        self.setLevel(logging.DEBUG)
+        # Our custom argument
+        self.log_list = log_list
+
+    def emit(self, record):
+        # record.message is the log message
+        self.log_list.append(record.msg)
+
+
+class TestMeasurementFail(TestCase):
+    def setUp(self):
+        logging.basicConfig(level=logging.DEBUG, format='%(message)s')
+        logger = logging.getLogger()
+        logger.level = logging.DEBUG
+        self.log_list = []
+        self.handler = ListHandler(self.log_list)
+        logging.getLogger().addHandler(self.handler)
+
+    def tearDown(self):
+        logging.getLogger().removeHandler(self.handler)
+        print('Stopped logging')
+
+        Measurement.final_actions.clear()
+        Measurement.except_actions.clear()
+
+    def test_measurement_fail_message(self):
+        with self.assertRaises(RuntimeError):
+            with Measurement('measurement_fail') as msmt:
+                raise RuntimeError('help')
+
+        self.assertTrue(any('Measurement error RuntimeError(help)' in msg for msg in self.log_list))
+
+    def test_measurement_except_final_actions(self):
+        p_except = Parameter(initial_value=42, set_cmd=None)
+        p_final = Parameter(initial_value=41, set_cmd=None)
+
+        with self.assertRaises(RuntimeError):
+            with Measurement('measurement_fail') as msmt:
+                msmt.except_actions.append(partial(p_except, 43))
+                msmt.final_actions.append(partial(p_final, 40))
+
+                raise RuntimeError('help')
+
+        self.assertEqual(p_except(), 43)
+        self.assertEqual(p_final(), 40)
+
+    def test_measurement_except_final_actions_no_fail(self):
+        p_except = Parameter(initial_value=42, set_cmd=None)
+        p_final = Parameter(initial_value=41, set_cmd=None)
+
+        with Measurement('measurement_fail') as msmt:
```

Comment:
I think this is meant to be 'measurement_no_fail'

Reply:
yep